vmm/src/memory_manager.rs

```diff
@@ -1617,10 +1627,9 @@ impl MemoryManager {
             return Err(Error::InvalidResizeWithMemoryZones);
         }
 
-        let mut region: Option<Arc<GuestRegionMmap>> = None;
         match self.hotplug_method {
             HotplugMethod::VirtioMem => {
-                if desired_ram >= self.boot_ram {
+                if desired_ram > self.boot_ram {
```

Comment:
Okay, this was `desired_ram >= self.boot_ram` for a reason. With virtio-mem you cannot shrink below the boot RAM. And the reason the test was failing was that it was trying to go back to the boot RAM amount and the resize was being silently rejected.